research/eip-draft_token_reputation.md

```diff
@@ -0,0 +1,376 @@
+
+Note there are two existing EIPs that have somewhat of a similar approach but come with flaws from the perspective of generalizing an interface for the broadest set of use cases:
+
+- [EIP-5192](https://eips.ethereum.org/EIPS/eip-5192): Designed for non-transferrable "soulbound" tokens ("SBT"), EIP-5192 has a few features that do not align with the token reputation requirements:
+  - Lacks a way to initiate _both_ staking and unstaking actions through standardized method calls—it only provides a `locked` getter method, which is unneeded for many use cases.
+  - Lack of indexed event parameters needed for the metadata, such as the caller's address or block number.
+  - Potentially misleading nomenclature if used for non-SBT use cases, whereas "staking" is more applicable to not only reputation-based scenarios but is also widely adopted elsewhere.
+- [EIP-5753](https://eips.ethereum.org/EIPS/eip-5753): Although EIP-5753 is currently a draft and yet to be accepted, it (essentially) changes EIP-5192 with the following:
+  - Adds `lock` and `unlock` methods, which behave similar to the token reputation proposal's `stake` and `unstake` methods; however, `unlock` lacks the required method caller information, as do the `Lock` and `Unlock` events.
+  - Changes the EIP-5192 `locked` getter to a function named `getLocked` that returns an address instead of boolean; but, getters SHOULD NOT be required for a minimal interface as this assumes contract storage is used to track reputation, which is not always the case (e.g., off-chain metadata / DA based reputation tracking).
+
+These two EIPs also lack the emittance of a `block.number` within their events. This is a key component for enabling reputation in metadata, which is a primary yet OPTIONAL feature of this proposal. For reputation to dynamically change within the metadata, there MUST be block information in the event emittance for off-chain indexing as this is needed to materialize any metadata updates with block-related information. In other words, reputation is most often tracked with a start and end block number that bound the staking/unstaking activities.
+
+For reference, [OpenSea](https://docs.opensea.io/docs/metadata-standards#disable-trading-for-staked-or-locked-tokens) leverages both of the events noted in EIP-5192 and EIP-5753 to disable trading for staked/locked tokens. There are other accepted events by OpenSea, including the `Stake` and `Unstake` events outlined in this proposal, which _do not_ have an EIP associated with them. Although it is not necessarily the primary motivation of this token reputation EIP, it's been noted to demonstrate how marketplaces are using this EIP's event definitions today. In other words, it is true that the events outlined in this proposal **will disable/enable trading a token on marketplaces**; this proposal _already_ has real-world usage.
```

Comment:
Have we tested that OpenSea does pick up these events automatically? I would guess that they use the event signature to look for relevant events, and if we change the number of event parameters or parameter types the signature changes right? https://docs.soliditylang.org/en/latest/abi-spec.html#events